lib/jxl/encode.cc

```diff
@@ -965,13 +965,25 @@ jxl::Status JxlEncoder::ProcessOneEnqueuedInput() {
 
       size_t save_as_reference =
           input_frame->option_values.header.layer_info.save_as_reference;
-      if (save_as_reference >= 3) {
+      if (save_as_reference > 3) {
         return JXL_API_ERROR(
             this, JXL_ENC_ERR_API_USAGE,
-            "Cannot use save_as_reference values >=3 (found: %d)",
+            "Cannot use save_as_reference values >3 (found: %d)",
             static_cast<int>(save_as_reference));
       }
 
+      if (input_frame->option_values.cparams.patches != jxl::Override::kOff) {
+        patches_possible = true;
+      }
+      if (save_as_reference == 3) {
+        used_reference_3 = true;
+      }
+      if (patches_possible && used_reference_3) {
+        return JXL_API_ERROR(this, JXL_ENC_ERR_API_USAGE,
+                             "Cannot use save_as_reference value 3 when "
+                             "patches are enabled for any frame");
```

Comment:
I feel that "some" would be more precise than "any".
@mo271 could you help with wording?